File: src/main/java/com/assets/board/model/ib/IBPosition.java
package com.assets.board.model.ib;

import com.opencsv.bean.CsvBindByName;
import lombok.Data;

@Data
public class IBPosition {

    private Long id;

    @CsvBindByName(column = "Symbol")
    private String symbol;

    @CsvBindByName(column = "Description")
    private String description;

    @CsvBindByName(column = "Quantity")
    private Integer quantity;

    @CsvBindByName(column = "MarkPrice")
    private Double markPrice;

    @CsvBindByName(column = "PositionValue")
    private Double positionValue;

    @CsvBindByName(column = "OpenPrice")
    private Double openPrice;

    @CsvBindByName(column = "CostBasisPrice")
    private Double costBasisPrice;

    @CsvBindByName(column = "FifoPnlUnrealized")
    private Double fifoPnlUnrealized;

    @CsvBindByName(column = "HoldingPeriodDateTime")
    private String holdingPeriodDateTime;

    @CsvBindByName(column = "Weight")
    private Double weight;
}

File: src/main/java/com/assets/board/model/ib/IBPosition.kt
package com.assets.board.model.ib

import com.opencsv.bean.CsvBindByName
import lombok.Data

@Data
data class IBPosition (
    private val id: Long? = null,

    @CsvBindByName(column = "Symbol")
    private val symbol: String? = null,

    @CsvBindByName(column = "Description")
    private val description: String? = null,

    @CsvBindByName(column = "Quantity")
    private val quantity: Int? = null,

    @CsvBindByName(column = "MarkPrice")
    private val markPrice: Double? = null,

    @CsvBindByName(column = "PositionValue")
    private val positionValue: Double? = null,

    @CsvBindByName(column = "OpenPrice")
    private val openPrice: Double? = null,

    @CsvBindByName(column = "CostBasisPrice")
    private val costBasisPrice: Double? = null,

    @CsvBindByName(column = "FifoPnlUnrealized")
    private val fifoPnlUnrealized: Double? = null,

    @CsvBindByName(column = "HoldingPeriodDateTime")
    private val holdingPeriodDateTime: String? = null,

    @CsvBindByName(column = "Weight")
    private val weight: Double? = null
)
